### ISSUE TITLE
Xenial Auth package fails to start with systemd ADDRESS_FAMILIES permission denied

### ISSUE THREAD

**Issue by mnordhoff** [PowerDNS/pdns#8346]
- Program: Authoritative
 - Issue type: Bug report

### Short description
Upgrading pdns-server from 4.2.0\~rc2+master.689.g5c3eb8c11-1pdns.xenial to 4.2.0\~rc2+master.723.g62232bc32-1pdns.xenial, it fails to start. syslog gets this:

```
Sep 25 21:07:13 clover systemd[1]: Starting PowerDNS Authoritative Server...
Sep 25 21:07:13 clover systemd[4496]: pdns.service: Failed at step ADDRESS_FAMILIES spawning /usr/sbin/pdns_server: Permission denied
Sep 25 21:07:13 clover systemd[1]: pdns.service: Main process exited, code=exited, status=232/ADDRESS_FAMILIES
Sep 25 21:07:13 clover systemd[1]: Failed to start PowerDNS Authoritative Server.
Sep 25 21:07:13 clover systemd[1]: pdns.service: Unit entered failed state.
Sep 25 21:07:13 clover systemd[1]: pdns.service: Failed with result 'exit-code'.
Sep 25 21:07:14 clover systemd[1]: pdns.service: Service hold-off time over, scheduling restart.
Sep 25 21:07:14 clover systemd[1]: Stopped PowerDNS Authoritative Server.
```

and so forth

### Environment
 - Operating system: Ubuntu 16.04 Xenial x86-64
 - Software version: 4.2.0\~rc2+master.723.g62232bc32-1pdns.xenial
 - Software source: PowerDNS repository

### Steps to reproduce
1. Install an older version of Auth (possibly optional)
2. `apt update`
3. `apt full-upgrade`

### Expected behaviour
It starts.

### Actual behaviour
It doesn't. ;-)

### Other information
`journalctl` says the same thing as `syslog`, except in color.

**Comment by mnordhoff** [comment]
This is a duplicate of #4219 (oops) except it always worked for me before?

**Comment by mnordhoff** [comment]
Diff between `pdns.service` in 689 (on the left) and 723 (on the right):

```diff
$ diff -u <(systemctl cat pdns) pdns.service 
--- /dev/fd/63  2019-09-25 22:47:37.923795285 +0000
+++ pdns.service        2019-09-25 18:28:38.000000000 +0000
@@ -1,4 +1,3 @@
-# /lib/systemd/system/pdns.service
 [Unit]
 Description=PowerDNS Authoritative Server
 Documentation=man:pdns_server(1) man:pdns_control(1)
@@ -8,13 +7,17 @@
 
 [Service]
 ExecStart=/usr/sbin/pdns_server --guardian=no --daemon=no --disable-syslog --log-timestamp=no --write-pid=no
+User=pdns
+Group=pdns
 Type=notify
 Restart=on-failure
 RestartSec=1
 StartLimitInterval=0
+RuntimeDirectory=pdns
 
 # Sandboxing
-CapabilityBoundingSet=CAP_NET_BIND_SERVICE CAP_SETGID CAP_SETUID CAP_CHOWN CAP_SYS_CHROOT
+CapabilityBoundingSet=CAP_NET_BIND_SERVICE CAP_CHOWN
+AmbientCapabilities=CAP_NET_BIND_SERVICE CAP_CHOWN
 ProtectHome=true
 # ProtectSystem=full will disallow write access to /etc and /usr, possibly
 # not being able to write slaved-zones into sqlite3 or zonefiles.
```

I don't get why I'm only having problems **now**. I'm pretty sure I never edited the service before.

**Comment by mnordhoff** [comment]
Fixed by #8348. (systemd is magic.)
